docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,71 @@
+# Tenant Configuration and Routing of Notifications in the Marain.UserNotifications service
+
+## Table of definitions
+
+| Word               | Definition                                                                                    |
+| ------------------ | --------------------------------------------------------------------------------------------- |
+| Consumer           | A service that uses the Marain.UserNotifications service                                      |
+| User               | The individual who is using an application that consumes the Marain.UserNotifications service |
+| Delivery Channel   | A third party service used to send a notification. Eg. Twilio, Airship, Sendgrid etc.         |
+| Communication Type | The mechanism of the delivery of a notification eg. Sms, Email, WebPush                       |
+
+## Status
+
+Proposed
+
+## Context
+
+Delivering notifications through different communication types (email, sms, web push notifications etc) is necessary in an extendable notifications service. Organizations should be able to choose which delivery channels (twilio, sendgrid, airship etc) they want to use for the associated communication type(s) as these said delivery channel could have different features or costs. This allows a consumer of `Marain.UserNotifications` to optimise their costs by determining the specific delivery channel for each communication type.
+
+## Decision
+
+Currently the `Marain.UserNotifications` service does not have any notification delivery functionality and is only used as a repository for notifications. This proposal will see the addition of delivery channel infrastructure to intergrate with third parties including but not limited to Airship, Twilio, SendGrid. Existing functionality will not be removed.
```

Comment:
This part:

> "is only used as a repository for notifications"

doesn't seem right, because the management API does in fact provide delivery functionality. Currently, that delivery is only into the one and only delivery channel available. The idea of this ADR is that its reach is expanded to other mechanisms, but that's an expansion of the existing capability rather than a fundamental shift: before and after this change, the management API is the means by which a consuming app asks to get a message delivered to an end user.